ipfs-cluster-service/main.go

```diff
@@ -276,7 +277,16 @@ removal, upgrade state using this command, and restart every peer.
 		if c.Bool("debug") {
 			setupDebug()
 		}
-		return nil
+
+		if _, err := os.Stat(absPath); err != nil {
+			// lockfile dir not init? don't lock
+			return nil
+		}
+		return lock(absPath)
```

Comment:
I think we don't need to lock every time. Things like `cluster-service version` don't need locking. A future `cluster-service shutdown` will need to run while locked anyway. This means you'd lock at the start of the commands that need locking (`deamon`, `migrate`) and you can `defer unlock()`

Reply:
Also, calling checkErr() on a locking error will ensure only the error message is printed, right now, it prints all the help when locking fails, which makes it difficult to realize the error is printed at the very top.